packages/nuxt/src/components/runtime/client-component.ts

```diff
@@ -0,0 +1,19 @@
+import { defineAsyncComponent, defineComponent, h } from "vue";
+import type { AsyncComponentLoader} from 'vue'
+import { default as ClientOnly } from '#app/components/client-only'
+
+/*@__NO_SIDE_EFFECTS__*/
+export const createClientPage = (loader: AsyncComponentLoader) => {
+  const page = defineAsyncComponent(loader);
+
+  return defineComponent({
+    inheritAttrs: false,
+    setup() {
+      return () => h('div', [
+        h(ClientOnly, undefined, {
+          default: () => h(page)
```

Comment:
can we find a way to send down page props ?

Reply:
sure thing :+1: